neqo-transport/src/streams.rs

```diff
@@ -467,6 +467,9 @@ impl Streams {
                 .remote()
                 .get_integer(tparams::INITIAL_MAX_DATA),
         );
+
+        self.events.send_stream_creatable(StreamType::BiDi);
+        self.events.send_stream_creatable(StreamType::UniDi);
```

Comment:
This can't be done unconditionally as you might not be able to create these streams if the limit is zero.  Check that the transport parameter is non-zero here.

Given that, you should create a test case (in neqo-transport) that tests for these events not being generated when the initial values are 0.

I'd also like to see a test in neqo-transport for the case where the stream limits are non-zero, rather than tacking an assertion onto something in neqo-http3.